contracts/libraries/LockDataLibrary.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+
+library LockDataLibrary {
+    uint256 private constant OFFSET = uint256(keccak256("LockData"));
+
+    function push(IPoolManager.LockData storage self, address locker) internal {
+        uint96 length = self.length;
+        uint256 index = self.index; // only 96 bits, but we expand to 256 for clarity when using bitwise or below
+
+        unchecked {
+            uint256 indexToWrite = OFFSET + length;
+            /// @solidity memory-safe-assembly
+            assembly {
+                // note that the current index becomes the parent lock index
+                sstore(indexToWrite, or(shl(96, locker), index))
+            }
+            self.length = length + 1;
+            self.index = length;
+        }
+    }
+
+    function getLock(uint256 i) internal view returns (address locker, uint96 parentLockIndex) {
+        unchecked {
+            uint256 position = OFFSET + i;
+            /// @solidity memory-safe-assembly
```

Comment:
cant you do
```
assembly ('memory-safe') {
}
```

https://docs.soliditylang.org/en/v0.8.20/assembly.html#memory-safety

Reply:
😆  https://github.com/Uniswap/v4-core/pull/173#discussion_r1221865465

Reply:
oh the compiler recognizes the comments format too! this way is just friendly for linting tools that don't have support for the latest compiler version, but honestly it's been long enough that they now might and this could be a non-issue